src/Build/BuildCheck/Infrastructure/BuildEventsProcessor.cs

```diff
@@ -91,11 +90,18 @@ public TaskKey(BuildEventContext context)
     /// <summary>
     /// The method collects events associated with the used environment variables in projects.
     /// </summary>
-    internal void ProcessEnvironmentVariableReadEventArgs(string envVarName, string envVarValue, string file, int line, int column)
+    internal void ProcessEnvironmentVariableReadEventArgs(ICheckContext checkContext, string envVarName, string envVarValue, string file, int line, int column)
     {
         if (!_evaluatedEnvironmentVariables.ContainsKey(envVarName))
         {
-            _evaluatedEnvironmentVariables.Add(envVarName, (envVarValue, file, line, column));
+            _evaluatedEnvironmentVariables.Add(envVarName, (envVarValue, ElementLocation.Create(file, line, column)));
+
+            EnvironmentVariableCheckData checkData =
+               new(file,
+                   checkContext.BuildEventContext?.ProjectInstanceId,
+                   _evaluatedEnvironmentVariables);
+
+            _buildCheckCentralContext.RunEnvironmentVariableActions(checkData, checkContext, ReportResult);
```

Comment:
I suppose we now want to send the reads one by one - correct?

Then we should get rid of `_evaluatedEnvironmentVariables` alltogether and change the `EnvironmentVariableCheckData` to accomodate single read data